server/monitor/src/test/java/org/apache/accumulo/monitor/view/WebViewsTest.java

```diff
@@ -0,0 +1,151 @@
+   * punctuation will cause a 400 response code.
+   */
+  @Test
+  public void testGetTablesConstraintViolations() {
+    Response output = target("tables/f+o*o").request().get();
+    Assert.assertEquals("should return status 400", 400, output.getStatus());
+  }
+
+  /**
+   * Test path tables/{tableID} which passes constraints. On passing constraints underlying logic will be executed so we need to mock a certain amount of it.
+   * Note: to get the proper response code back, you need to make sure jersey has a registered MessageBodyWriter capable of serializing/writing the object
+   * returned from your endpoint. We're using a simple stubbed out inner class HashMapWriter for this.
+   *
+   * @throws Exception
+   *           not expected
+   */
+  @Test
+  public void testGetTablesConstraintPassing() throws Exception {
+    Instance instanceMock = EasyMock.createMock(Instance.class);
+    expect(instanceMock.getInstanceID()).andReturn("foo").anyTimes();
+    AccumuloServerContext contextMock = EasyMock.createMock(AccumuloServerContext.class);
+    expect(contextMock.getInstance()).andReturn(instanceMock).anyTimes();
+
+    mockStatic(Monitor.class);
+    expect(Monitor.getContext()).andReturn(contextMock).anyTimes();
+
+    mockStatic(Tables.class);
+    expect(Tables.getTableName(instanceMock, new Table.ID("foo"))).andReturn("bar");
+    replayAll();
+    org.easymock.EasyMock.replay(instanceMock, contextMock);
+
+    // Using the mocks we can verify that the getModel method gets called via debugger
+    // however it's difficult to continue to mock through the jersey MVC code for the properly built response.
+    // Our silly HashMapWriter registered in the configure method gets wired in and used here.
+    Response output = target("tables/foo").request().get();
+    Assert.assertEquals("should return status 200", 200, output.getStatus());
+    String responseBody = output.readEntity(String.class);
+    Assert.assertTrue(responseBody.contains("tableID=foo") && responseBody.contains("table=bar"));
+  }
+
+  /**
+   * Test minutes parameter constraints. When outside range we should get a 400 response.
+   */
+  @Test
+  public void testGetTracesSummaryValidationConstraint() {
+    // Test upper bounds of constraint
+    Response output = target("trace/summary").queryParam("minutes", 5000000).request().get();
+    Assert.assertEquals("should return status 400", 400, output.getStatus());
+
+    // Test lower bounds of constraint
+    output = target("trace/summary").queryParam("minutes", -27).request().get();
+    Assert.assertEquals("should return status 400", 400, output.getStatus());
+  }
+
+  /**
+   * Silly stub to handle MessageBodyWriter for Hashmap. Registered in configure method and auto-wired by Jersey.
+   */
+  public static class HashMapWriter implements MessageBodyWriter<HashMap> {
```

Comment:
Should specify generic type parameters here on HashMap (and also line 140 and 145) for type safety and to avoid introducing new warnings.

Reply:
Reminder of these raw types warnings.